javascript/challenges/graphs/graph.js

```diff
@@ -0,0 +1,199 @@
+
+
+    while (queue.length) {
+      //const currentNode = queue.shift();
+      let currentNode;
+      for (let i = 0; i < queue.length; i++) {
+        currentNode = queue[i];
+        queue[0] = queue[1];
+        --queue.length;
+      }
+      //capture the value of first item in the array
+      //remove the value from the queue
+      //move every other item the array one position forward with a loop
+
+      //get all the neighbors of the node that i took off the queue
+      const neighbors = this.getNeighbors(currentNode);
+      for (const neighbor of neighbors) {
+        const neighborNode = neighbor.vertex;
+
+        //if the set has the node that i'm looking for
+        //if (visitedNodes.has(neighborNode)) {
+        for (let i = 0; i < visitedNodes.length; i++) {
+          if (visitedNodes[i] === neighborNode) {
+            console.log(`hitting if statement`);
+            //stop looking at this node and go to the next one
+            break;
+          } else {
+            console.log(`hitting else statement`);
+            //otherwise I haven't been there. I need to add it to the set
+            visitedNodes[visitedNodes.length] = neighborNode;
+            //visitedNodes.add(neighborNode);
+            
+          }
+        }
+        //put it into the queue
+        //queue.push(neighborNode);
+        if (!queue[0]) {
+          queue[queue.length] = neighborNode;
+        } else {
+          queue[queue.length + 1] = neighborNode;
+        }
+      }
+
+    }
+    //return the Set so I have a list of nodes of where I've been
+    console.log(visitedNodes);
+    return visitedNodes;
+  }
+
+  dfs(startNode) {
+    //add node to set
+    const visitedNodes = new Set();
+    const traverseNeighbors = (node) => {
+      visitedNodes.add(node);
+      const neighbors = this.getNeighbors(node);
+      for (const edge of neighbors) {
+        if (!visitedNodes.has(edge.vertex)) {
+          traverseNeighbors(edge.vertex);
+        }
+      }
+    };
+    traverseNeighbors(startNode);
+    return visitedNodes;
+  }
+
+  pathTo(startNode, endNode) {
+    const stack = [];
+    const visitedNodes = new Set();
+    const parentPath = new Map();
+
+    stack.push(startNode);
+    visitedNodes.add(startNode);
+
+    while (stack.length) {
+      const currentNode = stack.pop();
+
+      if (currentNode === endNode) {
+        return parentPath;
+      }
+
+      const neighbors = this.getNeighbors(currentNode);
+
+      for (let neighbor of neighbors) {
+        const neighborNode = neighbor.vertex;
+
+        if (visitedNodes.has(neighborNode)) {
+          continue;
+        } else {
+          visitedNodes.add(neighborNode);
+        }
+
+        stack.push(neighborNode);
+        parentPath.set(neighborNode, currentNode);
+      }
+    }
+    console.log(parentPath);
+    return parentPath;
+  }
+  size() {
+    return this.adjacencyList.size;
+  }
+}
+
+module.exports = { Vertex, Edge, Graph };
```

Comment:
Hey Garret! When you have a chance to come back to this and work on the tests I will get this regraded for you!